tests/FSharp.Compiler.ComponentTests/EmittedIL/TupleElimination.fs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.ComponentTests.EmittedIL
+
+open Xunit
+open FSharp.Test.Utilities.Compiler
+
+module ``TupleElimination`` =
+
+    [<Fact>]
+    let ``Sequence expressions with potential side effects do not prevent tuple elimination``() =
+        FSharp """
+module TupleElimination
+open System.Runtime.CompilerServices
+
+[<MethodImpl(MethodImplOptions.NoInlining)>]
+let f () = 3
+
+[<MethodImpl(MethodImplOptions.NoInlining)>]
+let sideEffect () = ()
+
+let x () =
+    let a, b =
+        "".ToString () |> ignore
+        System.DateTime.Now |> ignore
+        "3".ToString () |> ignore
+        2, f ()
+    System.DateTime.Now |> ignore
+    a + b
+
+let y () =
+    let a, b, c =
+        let a = f ()
+        sideEffect ()
+        a, f (), f ()
+    a + b + c
+
+let z () =
+    let a, b, c =
+        let u, v = 3, 4
+        sideEffect ()
+        f (), f () + u, f () + v
+    a + b + c
+         """
+         |> compile
+         |> shouldSucceed
+         |> verifyIL [
+            
+// public static int x()
```

Comment:
nit: commented code should be removed

Reply:
The comment is there to show the C# representation of the IL below. Thought it would be helpful. Should I still remove it?